plugins/de.ovgu.featureide.fm.core/src/de/ovgu/featureide/fm/core/job/SliceFeatureModel.java

```diff
@@ -104,10 +117,19 @@ private IFeatureModel sliceTree(Collection<String> selectedFeatureNames, IFeatur
 		final IFeature root = m.getStructure().getRoot().getFeature();
 
 		m.getStructure().setRoot(null);
-		m.reset();
 
 		// set new abstract root
+		// needs to be done before resetting to get a unique ID
 		final IFeature nroot = factory.createFeature(m, FeatureUtils.getFeatureName(orgFeatureModel, StringTable.DEFAULT_SLICING_ROOT_NAME));
+
+		final long nextElementId = m.getNextElementId();
+		m.reset();
+
+		// keep the nextElementId so newly created abstract features get a unique ID
+		if (m instanceof FeatureModel) {
```

Comment:
Using the implementation without the interface seems a bit hacky. It is probably needed here but should be at least documented. It is likely that the deletion will not work for attributed feature models because of that. Ideally, you would test this and document this in an issue if it is not working, by already pointing to this location.

Reply:
@sjvie @Raixxor What are your thoughts? Did you follow that advice?

Reply:
They are testing your point and will file an issue when they encounter a problem.

Reply:
We tested deleting features in extended feature models (as found in the examples) and it seems like that works just fine.
Are there other (attributed) feature models or does this suffice?

Reply:
Sounds great. You could try Velvet or UVL models, as they might also come with their own format. However, if it works with some other models this is already a good indicator.